src/Model/Table/SentencesTable.php

```diff
@@ -944,7 +944,28 @@ public function saveNewSentence($text, $lang, $userId, $correctness = 0, $basedO
             return false;
         }
 
-        return $this->saveWithDuplicateCheck($newSentence);
+        $sentence = $this->find('all')
+        ->where(['text' => $newSentence->text, 'lang' => $lang])
+        ->first();
+
+        // Duplicate sentence found
+        if ($sentence != null) {
+            // If sentence is orphan
+            if(empty($sentence->user_id)) {
+                $sentence->user_id = $userId;
+                return $this->save($sentence);
+            } else {
+                // If sentence is owned by spammer/inactive user, and you are an advanced contributor (or higher)
+                if($this->setOwner($sentence->id, $userId, CurrentUser::get('role'))) {
+                    $sentence->user_id = CurrentUser::get('id');
```

Comment:
I think you could change `setOwner` to return a sentence entity instead of just a boolean:
https://github.com/Tatoeba/tatoeba2/blob/67fd7ccb648f8d23d8e4daaa7dd8b568a7e8deac/src/Model/Table/SentencesTable.php#L1042-L1045
`save` could return `false` in case of a failure so throwing away its return value and always return `true` isn't really failproof.

Reply:
@AndiPersti Done! I had to change the function to [get the entire sentence object](https://github.com/Tatoeba/tatoeba2/pull/2598/files#diff-43e566d2d517ab1d9e32da7a397e04ba620afa5dc3e614cd59415cc9888f055eR1058), but I think it feels more proper this way. 

I also made it so that if the sentence is duplicate, [it is also marked OK](https://github.com/Tatoeba/tatoeba2/pull/2598/files#diff-43e566d2d517ab1d9e32da7a397e04ba620afa5dc3e614cd59415cc9888f055eR1058). What do you think of this code?

Its pretty late here, so I'll try to think about the other comment tomorrow. And thanks for the neat review! :)

Reply:
> I also made it so that if the sentence is duplicate, it is also marked OK. What do you think of this code?

@HelioStrike Based on the comments from Alan, it would be best to remove this code from this PR and to not start anything until we've discussed this a bit more. We need to find a consensus on if/when to add automatically an "OK" review.